qiskit/pulse/schedule.py

```diff
@@ -679,6 +682,19 @@ def assign_parameters(self,
 
         return self
 
+    def _unroll_param_dict(self, value_dict):
+        unrolled_value_dict = {}
+        for (param, value) in value_dict.items():
+            if isinstance(param, ParameterExpression):
+                unrolled_value_dict[param] = value
+            if isinstance(param, ParameterVector):
+                if not len(param) == len(value):
+                    raise PulseError('ParameterVector {} has length {}, which '
+                                     'differs from value list {} of '
+                                     'len {}'.format(param, len(param), value, len(value)))
+                unrolled_value_dict.update(zip(param, value))
+        return unrolled_value_dict
+
```

Comment:
This duplicates `QuantumCircuit._unroll_param_dict`. Can we consolidate these, maybe as a class method on `ParameterVector`?

Reply:
Sure! I will do it.

Reply:
@kdk the new method in `ParameterVector` will be:

```
def unroll_param_dict(self, values):
    """ Returns a dictionary with 'values' assigned to the parameter vector
    """
    unrolled_value_dict = {}
    if len(self._params) != len(values):
        raise CircuitError('ParameterVector {} has length {}, which '
                            'differs from value list {} of '
                            'len {}'.format(self._params, len(self._params), values, len(values)))
    unrolled_value_dict.update(zip(self._params, values))
    return unrolled_value_dict
```

and I can use the function as follows:

```
import qiskit as qk
from qiskit.circuit import ParameterVector

parameters = ParameterVector('θ', length = 3)
c = qk.QuantumCircuit(2)
c.rx(parameters[0],0)
c.rx(parameters[1],0)
c.rx(parameters[2],1)
c.draw()

b = c.assign_parameters(parameters.unroll_param_dict([1,2,3]))
```
However, the function `_unroll_param_dict(self, value_dict)` allowed us to evaluate `ParameterExpression` and `ParameterVector` at once, with the new method I should merge dictionaries in `assign_parameters`.
The next example shows this better:

**Before**
parameter_2 = Parameter("β")
c.rx(parameter_2, 1)
b = c.assign_parameters({parameters: [1,2,3], parameter_2: 4})

**With method in ParameterVector**
parameter_2 = Parameter("β")
c.rx(parameter_2, 1)
dict1 = parameters.unroll_param_dict([1,2,3])
dict1.update({parameter_2: 4})
b = c.assign_parameters(dict1)

Do you think that we should proceed with this approach or have another suggestion for where the function `_unroll_param_dict` should be or just using the last approach @lcapelluto?

Reply:
Good point. I don't think this makes sense to have as a public method on `ParameterVector` and you're right that the example code ends up more complicated (we want to preserve the ability for users to pass a vector and have `assign_parameters` unroll it internally). In this case, let's leave the duplication for now and we can revisit in the future.

Reply:
@taalexander do I need to do something else or this solution does not work?